numpy/core/src/umath/_scaled_float_dtype.c

```diff
@@ -456,6 +457,225 @@ init_casts(void)
 }
 
 
+/*
+ * We also wish to test very simple ufunc functionality.  So create two
+ * ufunc loops:
+ * 1. Multiplication, which can multiply the factors and work with that.
+ * 2. Addition, which needs to use the common instance, and runs into
+ *    cast safety subtleties since we will implement it without an additional
+ *    cast.
+ *
+ * NOTE: When first writing this, promotion did not exist for new-style loops,
```

Comment:
I had to think about what you meant. Maybe more explicit as
```
TODO: once promotion exists, this loop can be removed, since promotion 
from sfloat to double and vice versa should allow using the regular double * double
multiplication loop.
```

Reply:
For now, would it make sense to instead register the existing double-double multiplication loop? Conceptually, that is perhaps a bit cleaner than using promotion?

Reply:
Should just delete the comment again in the next PR (or just skip it).  I did mean that we can also use promotion to sfloat to implement `double * sfloat` via `sfloat * sfloat`. (Not the other way around, which also makes sense, but requires _wrapping_ of `double * double` which I don't have yet.)

Reply:
Yes, the wrapping would be useful for units, e.g., first convert degrees to radians and then call the trig function.